Merging feature random greeting into develop

gersonhiggins opened the pull request:
Create a new Rails app called 'hello-rails-react'.
Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
Verify that the jsbundling-rails gem is added to your Gemfile.
Initialize your project with Git.
Install needed node dependencies as described in the previous lesson's tutorial.
Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
Make sure that your project has a Postgres database set up.
Create a table for storing your messages and create 5 different greetings.
Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
Create a static view that will be the root of your app.
Create the App component with react-router and render it in your static view.
Create the Greeting component that will display a greeting. Set it up as a route in your App component.
Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
Display the random greeting in your Greeting component.

gersonhiggins committed:
functionality

gersonhiggins committed:
linters

gersonhiggins committed:
linters fix

gersonhiggins committed:
license and readme

gersonhiggins committed:
requested changes

gersonhiggins committed:
seeds

Whoistolu reviewed (changes requested):
Hi @gersonhiggins 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._
- [ ] [REQUIRED] I am recommending these changes here because it is not present in your file changes. You have a great implementation here @gersonhiggins , however, you seem to have forgotten a part of the project requirement that states that you should Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting. Let's get this implemented 👍🏽 
- [ ] [REQUIRED] It will be nice to create a proper `Greeting` component as stated by the project requirement instead of creating `home_controller.js` and you need to create a component directory where we will create the `App` and `Greeting` components. 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gersonhiggins committed:
add store

gersonhiggins committed:
Merge branch 'feature/random_greeting' of https://github.com/gersonhiggins/my_new_rails_app into feature/random_greeting

gersonhiggins committed:
linters fix

addod19 reviewed (changes requested):
## Hi @gersonhiggins, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Nice job adding the redux functionality :+1: 

 ## Required Changes ♻️
_Check the comments under the review._
**NOTE:**: I'm leaving the comments here because the component directory can not be found in the file changes
- [ ] It would be better if we create the `component` directory where we will create our `App` and `Greeting` components, without these components we can not display the greetings stored in the database. Here is a [link](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md) to the project requirements, kindly take some time and complete the project and ensure the project works as expected before submitting another code review request.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gersonhiggins commented:
@addod19 What do you mean, greeting are getting displayed

gersonhiggins commented on app/javascript/controllers/app.js:
```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { setGreeting } from '../redux/actions/greetingActions';
+
+function App() {
+  const message = useSelector((state) => state.greeting);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    // Make an HTTP request to fetch the random greeting
+    fetch("/random_greeting")
+      .then((response) => response.json())
+      .then((data) => {
+        dispatch(setGreeting(data.greeting));
+      })
+      .catch((error) => console.error("Error fetching data:", error));
+  }, [dispatch]);
+
+  return (
+    <div>
+      <h1>Hello World!</h1>
+      <p>{message}</p>
+    </div>
+  );
+}
+
+export default App;
```
here is is @addod19

gersonhiggins commented:
@addod19 do i have to call it greeting....

gersonhiggins committed:
requested changes

addod19 commented:
> @addod19 What do you mean, greeting are getting displayed

@gersonhiggins, sorry for the misunderstanding. I added a link to the review, I added it because it will help you understand what I'm trying to say. It's important we follow the project requirements, In your project I do not see a greeting component. Here is a screenshot of some parts of the project requirements, kindly compare it to what you have done. You decided to create the App component in the `controller` directory of the `app/javascript` directory, for that the requirements did not specify that. However, the rest of the requirements should be followed
![Screenshot from 2023-10-24 22-48-36](https://github.com/gersonhiggins/my_new_rails_app/assets/46388113/5375f5ec-f29b-418b-80c7-77790f4bc457)

gersonhiggins commented:
> > @addod19 What do you mean, greeting are getting displayed
> 
> @gersonhiggins, sorry for the misunderstanding. I added a link to the review, I added it because it will help you understand what I'm trying to say. It's important we follow the project requirements, In your project I do not see a greeting component. Here is a screenshot of some parts of the project requirements, kindly compare it to what you have done. You decided to create the App component in the `controller` directory of the `app/javascript` directory, for that the requirements did not specify that. However, the rest of the requirements should be followed ![Screenshot from 2023-10-24 22-48-36](https://user-images.githubusercontent.com/46388113/277826365-5375f5ec-f29b-418b-80c7-77790f4bc457.png)

gersonhiggins commented:
@addod19 yes but did i manage to fullfith the past reviewer request?... if his fail you should just said that there is no greeting component and keep going or just contact me becuase its only change a file name.

Whoistolu commented:
> @addod19 yes but did i manage to fullfith the past reviewer request?... if his fail you should just said that there is no greeting component and keep going or just contact me becuase its only change a file name.

Hello @gersonhiggins 
Yes, you managed to fix some of the changes but did not read the second recommendation. 
@addod19 was putting emphasis on my second recommendation.

Sorry about this. Please, make all the recommended fixes before asking for another review.

AbayomiOlaoye commented on app/javascript/application.js:
```diff
@@ -1,3 +1,15 @@
 // Entry point for the build script in your package.json
 import "@hotwired/turbo-rails"
 import "./controllers"
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import store from './redux/store';
+import App from './components/greeting';
+
+ReactDOM.render(
+  <Provider store={store}>
+    <App />
+  </Provider>,
+  document.getElementById('root')
+);
```
- [ ] As the previous reviewer indicated your choice of implementing the `React Router` here and advised that you continue with other requirements, I am also unable to access your `Greetings` as the page seems not to be found. Kindly check the project requirements and crosscheck your code to be sure your app opens. Attached is a  screenshot of my when I start the server after setting up the database and seeding some messages in.

![image](https://github.com/gersonhiggins/my_new_rails_app/assets/99715640/8f1d8501-7dee-4a3b-bf06-817ea04ee785)

AbayomiOlaoye reviewed (changes requested):
## STATUS: CHANGE REQUIRED ♻️ 

Hi @gersonhiggins :wave:  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![so_close](https://camo.githubusercontent.com/0c690ab997907a8c79afe7e695c891f6398cdbef93eaf8ce19f652ff321b54f8/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f744b797263304e546c4563475764546870342f67697068792e676966)


 ## Highlights
- Good PR message ✅ 
- Great commit messages ✅ 
- No linter error ✅ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] It will be very helpful and a good practice to write descriptive Pull Request title as this helps to ensure clarity and understanding of your code to viewers, visitors, etc.
![image](https://github.com/gersonhiggins/my_new_rails_app/assets/99715640/1d20582d-0c34-4876-bb22-2c2e8f1581fb)


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gersonhiggins commented:
![image](https://github.com/gersonhiggins/my_new_rails_app/assets/128397949/e1ac8a05-c258-4111-8b17-882be651d17e)
@AbayomiOlaoye working for me

gersonhiggins committed:
requested changes

MahmoudBakr23 reviewed (approved):
Hi @gersonhiggins 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done working on the requested changes from the previous reviewers so far. The app is complete. 👍 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
